### PR TITLE
zeal: unstable-2021-12-25 → unstable-2022-10-02

### PR DESCRIPTION
**Pull request by fabianhjr**
Change of note: qt5 → qt6

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabianhjr** [comment]
image

**Comment by peterhoeg** [comment]
Why the move to qt6? I'm not saying not to do it, but what is the reason?

**Comment by fabianhjr** [comment]
In general because, Upstream added qt6 support on July ( https://github.com/zealdocs/zeal/pull/1420 )

Personally because qt 6.4.1 compiled out of the box with gcc12 and qt 5.15.7 didn't (checked, still the latest lgpl patch release) doesn't

**Comment by peterhoeg** [comment]
The issue with qt6 is that the applications look like crap under KDE because the theme isn't picked up automatically. I'm much more in favour of doing something like this instead:
```nix
{
  zeal-qt5 = libsForQt5.callPac......;
  zeal-qt6 = qt6Packages.callPac......;
  zeal = zeal-qt5;
}
```
At least people have a choice between TheNewHotness and Doesn'tLookLikeTheRedHeadedStepChild.

**Comment by fabianhjr** [comment]
Oh, I wasn't aware of that issue. Will amend the merge request like that.

File: pkgs/data/documentation/zeal/default.nix [hunk 1]
```diff
@@ -22,10 +34,10 @@ mkDerivation rec {
       -e 's@^project.*@project(Zeal VERSION ${version})@'
   '';
 
-  nativeBuildInputs = [ cmake extra-cmake-modules pkg-config ];
+  nativeBuildInputs = [ cmake extra-cmake-modules pkg-config wrapQtAppsHook ];
 
   buildInputs = [
-    qtbase qtimageformats qtwebengine qtx11extras
```
Comment (fabianhjr) [inline pkgs/data/documentation/zeal/default.nix:28]: `qtx11extras` seems unneeded and is unavailable on qt6 however unsure if it is needed on other platforms.

**Comment by fabianhjr** [comment]
Only tested zeal-qt6 on GNOME 43, could you test zeal-qt5?

**Comment by peterhoeg** [comment]
Have a look at https://github.com/NixOS/nixpkgs/pull/202131 for how to deal with the differences between qt5 and qt6.

**Comment by fabianhjr** [comment]
Oh right

```
(if isQt5 then [
    qtlocation
  ]
```

That is very useful.

File: pkgs/data/documentation/zeal/default.nix [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
+, wrapQtAppsHook
+, xcbutilkeysyms
+}:
 
-mkDerivation rec {
+let
+  isQt5 = lib.versions.major qtbase.version == "5";
+in stdenv.mkDerivation rec {
   pname = "zeal";
   version = "0.6.999";
```
[the remark this answers is not in the document]
Comment (peterhoeg) [reply pkgs/data/documentation/zeal/default.nix:22]: Apart from that, good to go. Tested both on x64_86 linux - both working fine (except for qt6 looking ugly as expected).

[the remark this answers is not in the document]
Comment (fabianhjr) [reply pkgs/data/documentation/zeal/default.nix]: Done

**Pull request merged**